modules/text2vec-openai/vectorizer/objects.go

```diff
@@ -21,21 +28,42 @@ import (
 	libvectorizer "github.com/weaviate/weaviate/usecases/vectorizer"
 )
 
+const (
+	MaxBatchTime       = 40 * time.Second
+	MaxObjectsPerBatch = 2000 // https://platform.openai.com/docs/api-reference/embeddings/create
+)
+
+type batchJob struct {
+	texts      []string
+	tokens     []int
+	ctx        context.Context
+	wg         *sync.WaitGroup
+	errs       map[int]error
+	cfg        moduletools.ClassConfig
+	vecs       [][]float32
+	skipObject []bool
+}
+
 type Vectorizer struct {
 	client           Client
 	objectVectorizer *objectsvectorizer.ObjectVectorizer
+	jobQueueCh       chan batchJob
 }
 
 func New(client Client) *Vectorizer {
-	return &Vectorizer{
+	vec := &Vectorizer{
 		client:           client,
 		objectVectorizer: objectsvectorizer.New(),
+		jobQueueCh:       make(chan batchJob),
```

Comment:
maybe it should be buffered channel to more or less maintain order of incoming batch requests. now next job will be selected randomly out of everyone waiting to get to the queue

Reply:
Pretty sure that channels are first in, first out - only the blocking behaviour is different for buffered vs unbuffered. Or am I missing something?

Reply:
That is correct, provided channel has some storage (buffer) to keep elements in. That one has size of 0, so all elements wait to be pushed to the channel until worker is ready to fetch one. AFAIK elements are then selected randomly, so new one can be fetched before the one waiting for e.g. 30s already.